src/CodingContracts.ts

```diff
@@ -95,8 +51,8 @@ export type ICodingContractReward =
  * The player receives a reward if the problem is solved correctly
  */
 export class CodingContract {
```

Comment:
Can't the `CodingContractTypes[type]` be on the `CodingContract`? It would remove checking for it being undefined or assuming it's state type to be `unknown`.

Reply:
It took me a second, but I see what you mean now: Passing in the `CodingContractType` object directly, instead of the string `type` and needing to look it up.

Normally, that would be a better design, but I know why it's this way. The `CodingContract` class is serialized as part of the savegame, and this makes it both smaller (only the name is saved) and also more future-proof (if we need to bugfix a cct, that bugfix actually applies to existing saves, instead of the bugged version being persisted in old ccts).